scripts/ingests/utils.py

```diff
@@ -721,26 +741,57 @@ def ingest_parallaxes(db, sources, plxs, plx_errs, plx_refs, save_db=False, verb
         # If no previous measurement exists, set the new one to the Adopted measurement
-        # TODO: figure out adopted in cases of multiple measurements.
         adopted = None
+        duplicate = False
         source_plx_data = db.query(db.Parallaxes).filter(db.Parallaxes.c.source == db_name).table()
+        # TODO: Write a function to deal with setting adopted flag
         if source_plx_data is None or len(source_plx_data) == 0:
+            # if there's no other measurements in the database, set new data Adopted = True
             adopted = True
+            duplicate = False
+            old_adopted = None
+        elif len(source_plx_data) > 0:  # plx data already exists
+            for plx_data in source_plx_data:
+                if plx_data['reference'] == plx_refs[i]:  # check to see if other measurement is a duplicate of the new data
+                    duplicate = True
+                    verboseprint("Duplicate measurement\n", plx_data)
+                if plx_data['adopted'] == 1:  # check if another measurement is adopted
+                    old_adopted = plx_data
+            if not duplicate:
+                verboseprint("!!! Another Proper motion measurement exists,")
+                if verbose:
+                    source_plx_data.pprint_all()
+
+                # if errors of new data are less than other measurements, set Adopted = True.
+                if plx_errs[i] < min(source_plx_data['parallax_error']):
+                    adopted = True
+                    # unset old adopted
+                    if old_adopted:
+                        db.Parallaxes.update().where(and_(db.Parallaxes.c.source == old_adopted['source'],
+                                                             db.Parallaxes.c.reference == old_adopted['reference'])).\
+                            values(adopted=False).execute()
+                        old_adopted_data = db.query(db.Parallaxes).filter(and_(db.Parallaxes.c.source == old_adopted['source'],
+                                                                          db.Parallaxes.c.reference == old_adopted['reference'])).table()
+                        verboseprint("Old adopted measurement unset\n", old_adopted_data)
+
+                verboseprint("The new measurement's adopted flag is:", adopted)
+
         else:
-            print("\nOTHER PARALLAX EXISTS, adopted = None")
-            print(source_plx_data)
+            raise RuntimeError("Unexpected state")
 
-        # Construct data to be added
-        parallax_data = [{'source': db_name,
+        if not duplicate:
+            # Construct data to be added
+            parallax_data = [{'source': db_name,
                           'parallax': str(plxs[i]),
                           'parallax_error': str(plx_errs[i]),
                           'reference': plx_refs[i],
                           'adopted': adopted}]
+
         verboseprint(parallax_data, verbose=verbose)
```

Comment:
You shouldn't need to convert `plx[i]` and similar to strings; leaving them as floats should be fine and would avoid issues were you may have values that can't become strings.

Reply:
If we don't convert to strings, it adds numbers to the values. e.g, 45.23 becomes something like 45.23000123. But since the datatype is float in the database, it is ingested as a float even when I convert to a string. So I prefer to do the string thing so that we don't end up with the extra decimals.